src/main/java/oscar/command/Command.java

```diff
@@ -0,0 +1,68 @@
+package oscar.command;
+
+import java.time.format.DateTimeFormatter;
+
+import oscar.essential.Storage;
+import oscar.essential.TaskList;
+import oscar.exception.OscarException;
+
+/**
+ * Abstract class for various Oscar commands.
+ */
+public abstract class Command {
+    static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private final boolean isExit;
+
+    /**
+     * Constructor for command subclasses.
+     *
+     * @param isExit Whether command is an exit command.
+     */
+    public Command(boolean isExit) {
+        this.isExit = isExit;
+    }
+
+    /**
+     * Validates the provided task number.
+     *
+     * @param details Task number as a string.
+     * @param tasks ArrayList of tasks.
+     * @return Integer task number.
+     * @throws OscarException Failure to validate task number.
+     */
+    public int validateInt(String details, TaskList tasks) throws OscarException {
+        if (details.isEmpty()) {
```

Comment:
Good to see that single statement conditionals are wrapped by curly brackets still!